numpy/core/tests/test_umath.py

```diff
@@ -2918,8 +2917,25 @@ def test_rint_big_int():
     # Rint should not change the value
     assert_equal(val, np.rint(val))
 
-
 def test_signaling_nan_exceptions():
     with assert_no_warnings():
         a = np.ndarray(shape=(), dtype='float32', buffer=b'\x00\xe0\xbf\xff')
         np.isnan(a)
+
+def test_any_as_bool():
+    """ Test if any return bool as desired"""
+    a = np.any(np.zeros(2,dtype='O'))
+    b = np.any(np.ones(2,dtype='O'))
+    c = np.any(np.array([1,0],dtype='O'))
+    assert_equal(a, False)
+    assert_equal(b, True)
+    assert_equal(c, True)
+
+def test_all_as_bool():
+    """ Test if all return bool as desired"""
+    a = np.all(np.zeros(2,dtype='O'))
+    b = np.all(np.ones(2,dtype='O'))
+    c = np.all(np.array([1,0],dtype='O'))
+    assert_equal(a, False)
+    assert_equal(b, True)
+    assert_equal(c, False)
```

Comment:
Tests need some PEP fixes (spaces), but mostly I am missing a test for the `axis` kwarg, since that should return boolean arrays.

Reply:
Sure, can add that test for axis as well.